### ISSUE TITLE
V2 Missing Type Definitions - Pie, Bar

### ISSUE THREAD

**Issue by jcmoore0** [recharts/recharts#2319]
Team - The V2 is looking great!  
I found some issues with a few types.

## Bug with Typing 
1.  Bar Label - does not support custom Label renderer
2.  LabelProps - not exported  
3.  ToolTipprops - not exported
4.  Pie - onMouseEnter - does not support RechartsFunction  "(data, index, e)"

## Behavior
```
<Bar
  label={<CustomLabel/>}   <-- ts linter error,  workaround by adding // @ts-ignore
```
## What is the expected behavior?
* No error

## 2.0.0-beta.7

Thanks!
John

**Comment by reifnotreef** [comment]
@jcmoore0 How has v2 been working for you over all?  
I'm having some issues on a production app with recharts and core-js, upgrading to v2 was recommended on another issue thread.

**Comment by jcmoore0** [comment]
@reifnotreef - So good so far.  Bars, lines and pies are working well.  
I find nearly 100% backwards compatibility except the types are not quite the same as v1.x @types/recharts.

The one breaking change I found was Axis tick scale = band calculates tick x and offset differently.  I am not sure if it is intention but this example will not align Quarters properly in V2.
[https://recharts.org/en-US/examples/BarChartWithMultiXAxis](https://recharts.org/en-US/examples/BarChartWithMultiXAxis)

I am very happy the team wrote in Typescript and is making recharts better.

I did see a PR today to remove core-js.
